### PR TITLE
docs(samples): Adding samples for image related operations

### PR DESCRIPTION
**Pull request by m-strzelczyk**
Samples for the [Create, delete, and deprecate custom images](https://cloud.google.com/compute/docs/images/create-delete-deprecate-private-images) page.

### CODE REVIEW

File: samples/snippets/tests/test_images.py [hunk 1]
```diff
@@ -43,6 +48,39 @@ def test_disk():
...
+
+@pytest.fixture()
+def autodelete_image(autodelete_image_name):
+    """
+    An image that will be deleted after the test is done.
+    """
+    src_img = get_image_from_family('debian-cloud', 'debian-11')
+    new_image = create_image_from_image(PROJECT, src_img.name, autodelete_image_name, 'debian-cloud',
+                                        storage_location='eu')
+    yield new_image
```
Comment (parthea) [inline samples/snippets/tests/test_images.py:81]: The delete appears to be missing. Should there be a corresponding `delete` for the image created using `create_image_from_image`?

File: samples/snippets/images/set_deprecation_status.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    return result
+
+
+def set_deprecation_status(
+    project_id: str, image_name: str, status: compute_v1.DeprecationStatus.State
+) -> NoReturn:
+    """
+    Modify the deprecation status of an image.
+
+    Note: Image objects by default don't have the `deprecated` attribute at all, until it's set.
```
Comment (parthea) [inline samples/snippets/images/set_deprecation_status.py:82]:
```suggestion
    Note: Image objects by default don't have the `deprecated` attribute at all unless it's set.
```

**Review by parthea — APPROVED** [review]
Added minor observations

File: samples/snippets/tests/test_images.py (hunk 1 above)
Comment (m-strzelczyk) [reply samples/snippets/tests/test_images.py:81]: This fixture uses `autodelete_image_name` fixture to get a name for the temporary image. The `autodelete_image_name` fixture does the deleting.

**Pull request merged**